R/fread.R

```diff
@@ -302,6 +302,7 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir(), tz="UTC")
   for (j in which(nzchar(colClassesAs))) {       # # 1634
     v = .subset2(ans, j)
     new_class = colClassesAs[j]
+    if (new_class %chin% c("POSIXct")) v[!nzchar(v)] = NA_character_ # as.POSIXct/as.POSIXlt cannot handle as.POSIXct("") correctly #6208
```

Comment:
are there any other classes where this might apply? is it possible for `new_class` to be missing? wondering why this isn't just `new_class == "POSIXct"`...

If it's really limited to POSIXct, maybe it's cleaner instead to use `as.POSIXct(replace(v, !nzchar(v), NA_character_)` below? Though that will induce a copy, right? WDYT about `"POSIXct" = {v[!nzchar(v)] = NA_character_; as.POSIXct(v)}`?

Reply:
It also does not work for `as.POSIXlt` since `as.POSIXct.character` calls `as.POSIXlt`

Reply:
OK got it, hence the as.POSIXlt in the comment too. I think that's good enough.